src/main/java/org/gridsuite/loadflow/server/service/LoadFlowWorkerService.java

```diff
@@ -0,0 +1,219 @@
+        LoadFlowParameters params = context.getParameters();
+        LOGGER.info("Run loadFlow...");
+        Network network = getNetwork(context.getNetworkUuid(), context.getOtherNetworksUuids(), context.getVariantId());
+        String provider = context.getProvider();
+
+        Reporter rootReporter = Reporter.NO_OP;
+        Reporter reporter = Reporter.NO_OP;
+        if (context.getReportContext() != null) {
+            String rootReporterId = context.getReportContext().getReportName() == null ? LOAD_FLOW_TYPE_REPORT : context.getReportContext().getReportName() + "@" + LOAD_FLOW_TYPE_REPORT;
+            rootReporter = new ReporterModel(rootReporterId, rootReporterId);
+            reporter = rootReporter.createSubReporter(LOAD_FLOW_TYPE_REPORT, String.format(LOAD_FLOW_TYPE_REPORT + " (%s)", provider), "providerToUse", provider);
+        }
+
+        CompletableFuture<LoadFlowResult> future = runLoadFlowAsync(network, context.getVariantId(), context.getProvider(), params, reporter, resultUuid);
+
+        LoadFlowResult result = future == null ? null : future.get();
+        if (context.getReportContext().getReportId() != null) {
+            reportService.sendReport(context.getReportContext().getReportId(), rootReporter);
+        }
+        return result;
+    }
+
+    private void cleanLoadFlowResultsAndPublishCancel(UUID resultUuid, String receiver) {
+        resultRepository.delete(resultUuid);
+        notificationService.publishStop(resultUuid, receiver);
+    }
+
+    private void cancelLoadFlowAsync(LoadFlowCancelContext cancelContext) {
+        lockRunAndCancelLF.lock();
+        try {
+            cancelComputationRequests.put(cancelContext.getResultUuid(), cancelContext);
+
+            // find the completableFuture associated with result uuid
+            CompletableFuture<LoadFlowResult> future = futures.get(cancelContext.getResultUuid());
+            if (future != null) {
+                future.cancel(true);  // cancel computation in progress
+            }
+            cleanLoadFlowResultsAndPublishCancel(cancelContext.getResultUuid(), cancelContext.getReceiver());
+        } finally {
+            lockRunAndCancelLF.unlock();
+        }
+    }
+
+    @Bean
+    public Consumer<Message<String>> consumeRun() {
+        return message -> {
+            LoadFlowResultContext resultContext = LoadFlowResultContext.fromMessage(message, objectMapper);
+            try {
+                runRequests.add(resultContext.getResultUuid());
+                AtomicReference<Long> startTime = new AtomicReference<>();
+
+                startTime.set(System.nanoTime());
+                LoadFlowResult result = run(resultContext.getRunContext(), resultContext.getResultUuid());
+                long nanoTime = System.nanoTime();
+                LOGGER.info("Just run in {}s", TimeUnit.NANOSECONDS.toSeconds(nanoTime - startTime.getAndSet(nanoTime)));
+
```

Comment:
You need something like : 
if (result.isOk()) {
                networkStoreService.flush(network);
            }
in this method

Reply:
Done